src/main/java/dk/aau/cs/io/TimedArcPetriNetNetworkWriter.java

```diff
@@ -295,10 +296,14 @@ private void appendArcs(Document document, DataLayer guiModel, Element NET) {
 	private void appendQueries(Document document, Element root) {
 		for (TAPNQuery query : queries) {
 			Element newQuery;
-			if (query.getCategory() == QueryCategory.LTL){
-			    newQuery = createLTLQueryElement(query, document);
-            } else {
-                newQuery = createCTLQueryElement(query, document);
+			if (query.getCategory() == QueryCategory.CTL){
+				newQuery = createCTLQueryElement(query, document);
+			} else if (query.getCategory() == QueryCategory.LTL) {
+                newQuery = createLTLQueryElement(query, document);
+            } else if(query.getCategory() == QueryCategory.HyperLTL) {
+			    newQuery = createHyperLTLQueryElement(query, document);
+            }else {
+				newQuery = createQueryElement(query, document);
```

Comment:
We do not use createQueryElement anymore, as all queries are converted to XML instead of text.

You should therefore remove the if (query.getCategory() == QueryCategory.CTL) statement (lines 299-300) and replace the else-statement (line 306) with newQuery = createCTLQueryElement(query, document);

Reply:
Fixed in https://github.com/TAPAAL/tapaal-gui/pull/39/commits/e0651b05f7e4c6329031295bd8540cab513b49f0